lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoFileSystem.java

```diff
@@ -207,4 +207,17 @@ void renameDirectory(Location source, Location target)
      */
     Set<Location> listDirectories(Location location)
             throws IOException;
+
+    /**
+     * Creates a temporary directory for the target path. The directory will be created
+     * using the (possibly absolute) prefix such that the directory can be renamed to
+     * the target path. The relative prefix will be used if required for the target path.
```

Comment:
This took me a few reads to understand.  I think it would be helpful to lead with something like:

> Create a temporary directory that can be renamed to the specified target path

I also think the "relative prefix" sentence could be improved.  Is it just a suggestion?  I ask becaue of "if required".

Reply:
The "if required" means that the target path requires a relative prefix because it doesn't support the absolute prefix. This method is only used for Hive and this relative prefix stuff is only there for ViewFS (it should probably also be used for HDFS encryption zones rather than skipping the temporary directory). I'll try improve the wording.